File: packages/SystemUI/tests/src/com/android/keyguard/KeyguardStatusViewTest.java
/*
 * Copyright (C) 2018 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License
 */

package com.android.keyguard;

import static org.mockito.Mockito.verify;

import android.test.suitebuilder.annotation.SmallTest;
import android.testing.AndroidTestingRunner;
import android.testing.TestableLooper;
import android.testing.TestableLooper.RunWithLooper;
import android.view.LayoutInflater;

import com.android.systemui.SysuiTestCase;
import com.android.systemui.util.Assert;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;

@SmallTest
@RunWithLooper
@RunWith(AndroidTestingRunner.class)
public class KeyguardStatusViewTest extends SysuiTestCase {

    @Mock
    KeyguardSliceView mKeyguardSlice;
    @Mock
    KeyguardClockSwitch mClockView;
    @InjectMocks
    KeyguardStatusView mKeyguardStatusView;

    @Before
    public void setUp() {
        Assert.sMainLooper = TestableLooper.get(this).getLooper();
        LayoutInflater layoutInflater = LayoutInflater.from(getContext());
        mKeyguardStatusView =
                (KeyguardStatusView) layoutInflater.inflate(R.layout.keyguard_status_view, null);
        org.mockito.MockitoAnnotations.initMocks(this);
    }

    @Test
    public void dozeTimeTick_updatesSlice() {
        mKeyguardStatusView.dozeTimeTick();
        verify(mKeyguardSlice).refresh();
    }

    @Test
    public void dozeTimeTick_updatesClock() {
        mKeyguardStatusView.dozeTimeTick();
        verify(mClockView).refresh();
    }

}

File: packages/SystemUI/tests/src/com/android/keyguard/KeyguardStatusViewTest.kt
package com.android.keyguard

import android.test.suitebuilder.annotation.SmallTest
import android.testing.AndroidTestingRunner
import android.testing.TestableLooper.RunWithLooper
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.android.systemui.R
import com.android.systemui.SysuiTestCase
import com.android.systemui.util.children
import com.google.common.truth.Truth.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@SmallTest
@RunWith(AndroidTestingRunner::class)
@RunWithLooper(setAsMainLooper = true)
class KeyguardStatusViewTest : SysuiTestCase() {

    private lateinit var keyguardStatusView: KeyguardStatusView
    private val mediaView: View
        get() = keyguardStatusView.findViewById(R.id.status_view_media_container)
    private val statusViewContainer: ViewGroup
        get() = keyguardStatusView.findViewById(R.id.status_view_container)
    private val clockView: ViewGroup
        get() = keyguardStatusView.findViewById(R.id.keyguard_clock_container)
    private val childrenExcludingMedia
        get() = statusViewContainer.children.filter { it != mediaView }

    @Before
    fun setUp() {
        keyguardStatusView = LayoutInflater.from(context)
                .inflate(R.layout.keyguard_status_view, /* root= */ null) as KeyguardStatusView
    }

    @Test
    fun setChildrenTranslationYExcludingMediaView_mediaViewIsNotTranslated() {
        val translationY = 1234f

        keyguardStatusView.setChildrenTranslationY(translationY, /* excludeMedia= */true)

        assertThat(mediaView.translationY).isEqualTo(0)

        childrenExcludingMedia.forEach {
            assertThat(it.translationY).isEqualTo(translationY)
        }
    }

    @Test
    fun setChildrenTranslationYIncludeMediaView() {
        val translationY = 1234f

        keyguardStatusView.setChildrenTranslationY(translationY, /* excludeMedia= */false)

        statusViewContainer.children.forEach {
            assertThat(it.translationY).isEqualTo(translationY)
        }
    }

    @Test
    fun setAlphaExcludeClock() {
        keyguardStatusView.setAlpha(0.5f, /* excludeClock= */true)
        assertThat(statusViewContainer.alpha).isNotEqualTo(0.5f)
        assertThat(mediaView.alpha).isEqualTo(0.5f)
        assertThat(clockView.alpha).isNotEqualTo(0.5f)
    }
}
